### PR TITLE
fix #3398 docs(project): Add commit formatting guidance

### LINKED ISSUE #3398: Update contributing.md to include FXA protocols
The FXA team has a lot of well thought out practices around commits, we should incorporate those into our contributing.md docs: https://github.com/mozilla/fxa/blob/main/CONTRIBUTING.md#git-commit-guidelines

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/EXP-256)
┆Issue Number: EXP-256

### PR DESCRIPTION
**Pull request by jaredlockhart**
Because

* We want to use fancy well defined commit formatting

This commit

* Adds rich well defined commit formatting guidance to our contributing docs

### CODE REVIEW

File: contributing.md [hunk 1]
```diff
@@ -37,10 +37,8 @@ do not require tests.
   - Fix any broken tests
   - Ensure coverage remains at 100% by adding new tests for any uncovered lines.
 
-1. When all tests and checks are passing, commit all your changes into a single commit and name
-the commit with a brief description of the change and `fixes #123` where `#123` is the issue number.
-You can add any additional descriptive information to the message body of the commit, but the issue
-number must be present in the first line of the commit.  Example: `Add new dashboard link fixes #123`
+1. When all tests and checks are passing, commit all your changes into a single commit and follow
+the [Git Commit GUidelines](#git-commit-guidelines)
```
Comment (LZoog) [inline contributing.md:41]: ~GUidelines~ Guidelines

Also I don't think this needs a new line.

File: contributing.md [hunk 2]
```diff
@@ -64,14 +62,67 @@ contains any additional descriptive information.
...
+- **perf**: A code change that improves performance
+- **test**: Adding missing tests
+- **chore**: Changes to the build process or auxiliary tools and libraries such as documentation
+  generation
+
+#### Scope
+
+One of the following:
+
+- **example scope**
```
Comment (jaredlockhart) [inline contributing.md:91]: Open to suggestions here, not sure what the granularity should be like? Like maybe

- project (project wide changes)
- dev (things that impact dev environments?)
- backend
- frontend
- ...others?

File: contributing.md [hunk 3]
```diff
@@ -64,14 +62,67 @@ contains any additional descriptive information.
...
+
+* Reason
+
+This commit
+
+* Change
+```
+
+Just as in the **subject**, use the imperative, present tense: "change" not "changed" nor "changes". Commits
+are expected to follow this format.
```
Comment (LZoog) [inline contributing.md:116]: No new line needed

**Review by LZoog — APPROVED** [review]
LGTM, almost identical to FxA's.

File: contributing.md (hunk 2 above)
Comment (LZoog) [reply contributing.md:90]: Oh not sure how I glossed over this section 😂

Project is definitely a good one. Something like `docs(nimbus-ui):` or `feat(nimbus-ui)` is pretty descriptive, though for BE changes I'm not sure what you'd call that, `feat(backend)` would maybe work there?

**Review by bbangert — APPROVED** [review]
LGTM!

**Comment by pdehaan** [comment]
👀 

> **Description**
> - don't capitalize first letter

Re: "fix #3398 docs(project): Add commit formatting guidance"

Shouldn't this be "add commit" instead of "Add commit"?

**Comment by jaredlockhart** [comment]
@pdehaan Ha! I'll have to remember that ;)

**Comment by pdehaan** [comment]
I vaguely recall FxA used to use [GitCop](https://gitcop.com/) to enforce git commit messages (it made it easier to generate a CHANGELOG file).

We've used a few other solutions in the past for similar things using Angular commit message styles. Not sure if there is a generic validator that we would be able to bend to our will to support our preferred `fixes #github_issue_number type(scope): Description` style.

I think I did a quick audit of the 7 existing PRs and found that none of them exactly match the style 100%.

**Comment by jaredlockhart** [comment]
Who needs GitCop when we have you @pdehaan ;D

**Comment by pdehaan** [comment]
Realtime GitHub Commit Message Linting as a Service.

**Comment by pdehaan** [comment]
https://commitlint.js.org/#/reference-rules satisfies most of the requirements, with the notable exception of the "fixes #github_issue_number" prefix.

**Pull request merged**